src/wallet/wallet.cpp

```diff
@@ -3865,7 +3871,7 @@ CWallet* CWallet::CreateWalletFromFile(const std::string walletFile)
     RegisterValidationInterface(walletInstance);
 
     CBlockIndex *pindexRescan = chainActive.Genesis();
-    if (!GetBoolArg("-rescan", false))
```

Comment:
IMO Yes. It's pure db utility. Either we move it to RPC or to a new wallet/db tool.